Add documentation build and links checking in CI.

uggla opened the pull request:
Sanity check:

* [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [X] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [X] Have you created/updated the relevant documentation page(s)?

Explanations:

@Keats, you told me in a comment that it was something you wanted to add to the CI. I did it as I could handle it and let you manage more complex stuffs.
* Add `zola build` and `zola check` to ensure documentation can be built and does not contain invalid links.
* Fix links in the current documentation that were broken.
Maybe I would have to open this PR on the master branch ? Let me know if you would rather this PR on the master branch.

uggla committed:
Add CI documentation checks

- Ensure documentation can be built.
- Check documentation links are valid.

uggla committed:
Fix failed links

uggla committed:
Fix more links into recent docs

uggla committed:
Reduce the thread pool size for link checking

- Try to see if it is more stable with less thread especially on systems
  with few ressources.

uggla committed:
Set thread pool size according to number of cpus

uggla committed:
Display number of cpu

uggla committed:
Reduce number of cpu and improve message

Keats commented on azure-pipelines.yml:
```diff
@@ -41,6 +41,24 @@ stages:
+          - script: |
+              cd docs && ..\target\debug\zola check
+            condition: eq( variables['Agent.OS'], 'Windows_NT' )
+            displayName: Check documentation links windows
```
That's no good, it would block the release of Zola if there is a dead link. I would be tempted to move `zola check` to GH actions actually.

uggla replied:
Hello @Keats, in fact that was the intention. :smile: 
To not pass the CI process if for any reason the documentation was broken or of boken links was introduced (Like the ones I corrected). Also it would check this on a regular basis and avoid documentation to deviate and keep it completely updated.
I like to put all checks in the CI, but if you think it annoying I can remove them from this PR.

Keats replied:
It's a good instinct but the link checker is not robust yet so that could cause me to comment out these lines in anger if I want to do a release and it's failing for whatever reason

uggla replied:
So I comment out the link check part.
I introduced nb_threads_per_cpu to configure the link_checker number of threads. As reducing the number of threads looks more stable at least on AzureDevOps. Maybe because there is some kind of limitation to traffic and especially number of simultaneous connections outgoing from AzureDevOps to avoid DDOS.

Keats commented on docs/config.toml:
```diff
@@ -8,5 +8,16 @@ insert_anchor_links = true
+# skip_anchor_prefixes = [
+#     "https://caniuse.com/",
+# ]
+
```
what's the reason for that change?

uggla replied:
In fact I did a configuration documentation copy / paste.
```
[link_checker]
# Skip link checking for external URLs that start with these prefixes
skip_prefixes = [
    "http://127.0.0.1",
]

# Skip anchor checking for external URLs that start with these prefixes
# skip_anchor_prefixes = [
#     "https://caniuse.com/",
# ]
```
I wanted to avoid broken links errors with `http://127.0.0.1` (there are a several errors in the documentation), but so far I don't have to use the "anchors skip part". So I commented it out.

uggla committed:
Allow to pass link_checker nb_threads_per_cpu

- nb_threads_per_cpu define the number of thread per cpu the
  link_checker will use to check links.
  Example : if your system is 2 cpus and nb_threads_per_cpu = 6 you will
  get 6 * 2 = 12 threads to check links.
- nb_threads_per_cpu default to 4.

uggla committed:
Update documentation about nb_threads_per_cpu

uggla committed:
Comment link checking part

uggla committed:
Fix wrong call of Linkchecker in tests

uggla committed:
Fix conf in config::tests::link_checker_nb_threads_per_cpu

Keats commented on azure-pipelines.yml:
```diff
@@ -41,6 +41,26 @@ stages:
+          - script: |
+              cd docs && ..\target\debug\zola build
+            condition: eq( variables['Agent.OS'], 'Windows_NT' )
+            displayName: Check documentation build windows
```
I like that!

Keats commented on components/config/src/config/link_checker.rs:
```diff
@@ -7,10 +7,15 @@ pub struct LinkChecker {
     pub skip_prefixes: Vec<String>,
     /// Skip anchor checking for these URL prefixes
     pub skip_anchor_prefixes: Vec<String>,
+    pub nb_threads_per_cpu: usize,
```
I don't think this should be configurable. Let's keep it at 4 for now and we'll see for configuration later on.

uggla replied:
So I come back to the static version

Keats commented on components/site/src/link_checking.rs:
```diff
@@ -119,9 +119,14 @@ pub fn check_external_links(site: &Site) -> Result<()> {
+        "Number of cpus: {}, using {} threads",
+        num_cpus::get(),
+        num_cpus::get() * nb_threads_per_cpu
+    );
```
we don't want to print that

uggla replied:
I have removed it.

Keats commented on docs/content/themes/codinfox-zola/index.md:
```diff
@@ -21,11 +21,11 @@ homepage = "https://svavs.github.io/"
 This theme places content first by tucking away navigation in a hidden drawer.
 
-* Built for [Zola](https://www.getzola.com)
+* Built for [Zola](https://www.getzola.org)
```
Those are generated by the https://github.com/getzola/themes repo, fixing it here will not prevent more bad permalinks on the next update.

Keats reviewed (commented):
Overall I like the build on CI but not sure about the rest. We can definitely lower the number of threads for the link checker though, in a more static way for now.

uggla committed:
Revert "Fix conf in config::tests::link_checker_nb_threads_per_cpu"

This reverts commit afccef5d983c99fb298add022ae40f025a5357f7.

uggla committed:
Revert "Fix wrong call of Linkchecker in tests"

This reverts commit b7ad0f4acfeb94c79fec87d8f7ef33a16bdaf6e5.

uggla committed:
Revert "Update documentation about nb_threads_per_cpu"

This reverts commit c0f3d3ee18a5bd0fb90759736cbdd4c13db93985.

uggla committed:
Revert "Allow to pass link_checker nb_threads_per_cpu"

This reverts commit 0e6e7fc85926922811e873f9a155cbc027e29f45.

uggla committed:
Remove message about cpu and thread used

uggla committed:
Fix comment about default threads
